src/config.js

```diff
@@ -82,13 +83,13 @@ const config = convict({
   mongoUri: {
     doc: 'URI for mongodb',
     format: String,
-    default: 'mongodb://127.0.0.1:27017/',
+    default: null,
```

Comment:
Removing default value so service will fail fast on startup if dependency missing.

Reply:
We have a slightly awkward way of the config currently provides dev defaults. Same again this is on my TODO list. Maybe something along the lines of `convict` and `dotenv` to provide a dev defaults in an env file committed to the repo. We are yet to look at this. But if we are removing defaults from this file it would be nice, IMO to also provide a way of setting envs for development?

Reply:
Generally this shouldn't be an issue since we set `MONGO_URI` at a platform/env level for all services

Reply:
Agree for default Platform injected stuff it's unlikely to be an issue.  But it does set a precedent for how config should be set for external values.

I've seen several teams (and me to be fair) caught out by not failing fast on missing config and only discovering an issue later when that config item is needed.

Reply:
For .env files you could also pass `--env-file=.env` to the node process too rather than rely on dotenv.  I guess not every team will have local sensitive config though so may not be required by default on the template.

Reply:
FYI this will need the `nullable: true,` adding

Reply:
> I've seen several teams (and me to be fair) caught out by not failing fast on missing config and only discovering an issue later when that config item is needed.

The convict `config.validate({ allowed: 'strict' })` can help here with failing fast

Reply:
> The convict config.validate({ allowed: 'strict' }) can help here with failing fast

Agreed, and should absolutely be used.  But if there's a default local dev value in there then the validation would still pass.

Reply:
Is `convict` even needed? What value does it brings over the simple `joi` based [approach](https://github.com/DEFRA/hapi-web-boilerplate/blob/master/server/config.js), with the ENV VARS optionally [documented](https://github.com/DEFRA/hapi-web-boilerplate/tree/master?tab=readme-ov-file#environment-variables) in the main README? Is it because it has type inference support?

Reply:
@davidjamesstone convict was added as a personal preference, but no reason not to swap it out for `Joi`

Reply:
I'd agree with that.  Our standards focus on the 12 factor app principals such as config should be external, but we haven't got anything prescribing how to achieve that and which libraries to use.

There's a mix in play across Defra.  A lot of Joi and a growing use of Convict (Probably due to these templates).

My view would be the template should just pick one and teams can swap it out if they don't like the choice.  It's a fairly lightweight change either way.

My personal choice is `<REDACTED>`